architecture/2025-02-user-defined-resource-type-feature-spec.md

```diff
@@ -0,0 +1,1361 @@
+
+Radius will continue to ship with core resource types including containers, auto-scalers, gateways, volumes, and secrets. It will have built-in logic for deploying these resources to various container platforms including Kubernetes, ECS, ACA, ACI, and CloudRun. In order to allow the creation of more abstract, application-oriented resource types, user-defined resource types will support the concept of child resources.
+
+Child resources are resources which get deployed when its parent resource is deployed. The lifecycle of the child resource is tied to that of the parent. These child resources can be a mix of core resource types and user-defined resource types. 
+
+> [!NOTE]
+>
+> Bicep has built-in support for child resources which was a primary reason Bicep was chosen over YAML.
+
+**User Experience**
+
+The platform engineer creates a resource type which has a child resource embedded within the resource type definition. 
+
+**`service-resource-type.bicep`**:
+
+```yaml
+resource MyCompany.App/service 'System.Resources/resourceTypes@2023-10-01-preview' = {
+  name: 'MyCompany.App/service'
+  description: 'The service resource type is a long-running process responding to HTTP requests over TLS.'
+  api: {
+    version: 'v1alpha1'
+    schema: {
+      properties: {
+        container-image: {
+          type: 'string'
+          description: 'Container image name'
+        }
+        ...
+      }
+    }
+  } 
+  // Child resource
+  resource proxy 'Applications.Core/containers@2023-10-01-preview' = {
+    name: 'proxy'
+    properties: {
+      container: {
+        image: 'envoy:latest'
+        ...
+      }
+    }
+  }
+  // Child resource; note the name is taken from the parent
+  resource ${parent.name} 'Applications.Core/containers@2023-10-01-preview' = {
+    name: ${parent.name}
+    properties: {
+      container: {
+        // Use impage property from the parent resource
+        image: ${parent.image}
+        ...
+      }
+    }
+  }
+}
+```
+
+When a developer creates a `MyCompany.App/service` resource, they will not need to have any awareness that an Envoy proxy is also created. This abstraction enables the platform engineer to easily inject sidecar containers. Envoy is shown in the example here, but other common sidecars are container security scanning tools such as Aqua Security and observability tools such as OpenTelemetry or Datadog. 
+
+The developer will, however, see that Envoy is deployed when they inspect the application graph since the child resources are standard Radius resources and will appear in the application graph.
```

Comment:
Does Radius support child resources as of now? Is there an example Radius Bicep to that?